### PR TITLE
fixing sp role assignment

### PR DESCRIPTION
**Pull request by erinborders**
Before this change, the service principal was just being assigned a reader role. Now it's being assigned a contributor role.

### CODE REVIEW

**Review by erinborders — COMMENTED** [review]
triggering integration tests

File: pkg/providers/azure.go [hunk 1]
```diff
@@ -164,8 +164,9 @@ func (sc *SetUpCmd) CreateServicePrincipal() error {
 
 func (sc *SetUpCmd) assignSpRole() error {
 	log.Debug("Assigning contributor role to service principal...")
+	contributorId := "b24988ac-6180-42a0-ab88-20f7382dd24c"
```
Comment (Tatsinnit) [inline pkg/providers/azure.go:167]: 💡 Nice, just an idea - I think this `guid` refer to this doc - https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#all might as well comment above the implementation for readers who might wonder what this guid translation leads to.

File: pkg/providers/azure.go [hunk 2]
```diff
@@ -164,8 +164,9 @@ func (sc *SetUpCmd) CreateServicePrincipal() error {
...
 	scope := fmt.Sprintf("/subscriptions/%s/resourceGroups/%s", sc.SubscriptionID, sc.ResourceGroupName)
-	assignSpRoleCmd := exec.Command("az", "role", "assignment", "create", "--role", "contributor", "--subscription", sc.SubscriptionID, "--assignee-object-id", sc.spObjectId, "--assignee-principal-type", "ServicePrincipal", "--scope", scope, "--only-show-errors")
+	assignSpRoleCmd := exec.Command("az", "role", "assignment", "create", "--role", contributorId, "--subscription", sc.SubscriptionID, "--assignee-object-id", sc.spObjectId, "--assignee-principal-type", "ServicePrincipal", "--scope", scope, "--only-show-errors")
```
Comment (Tatsinnit) [inline pkg/providers/azure.go:169]: 💡 sharing thoughts only for idea - Given that this change will now assign the `contributor` level role, do we need to make any further tool level read-me changes et. al. as this change is moving away from access/security policy of starting narrow and then go broad to broad.

**Review by Tatsinnit — APPROVED** [review]
☕️❤️ very cool, thank you os much for this change, I've added few comments just as improvement. Also, a thought where if we are changing the default to `contributor` level access which this mean we are broadening the access hence end - user should be very aware of this change or there will not be any ripple effect. 

looks good to me, but @davidgamero for more tool specific eyes for impact.

**Pull request closed without merging**